### PR TITLE
[ADLN] IT8659 Sio DTT porting

### PR DESCRIPTION
**Pull request by koktong-ong**
Ported IT8659 Sio (Super IO) and DTT (Dynamic Tuning Technology) for
ADL platform
Added CfgData to disable or enable the SIO feature

Signed-off-by: Ong Kok Tong <kok.tong.ong@intel.com>

### CODE REVIEW

File: Platform/AlderlakeBoardPkg/Library/Stage2BoardInitLib/SioChip.c [hunk 1]
```diff
@@ -0,0 +1,763 @@
+/** @file
+
+  Copyright (c) 2020 - 2022, Intel Corporation. All rights reserved.<BR>
```
Comment (kvparikh89) [inline Platform/AlderlakeBoardPkg/Library/Stage2BoardInitLib/SioChip.c:3]: This is a new file. Remove 2020 from header

File: Platform/AlderlakeBoardPkg/Library/Stage2BoardInitLib/SioChip.c [hunk 2]
```diff
@@ -0,0 +1,763 @@
...
+  IN  UINT8     Register,
+  OUT UINT8    *Value
+  )
+{
+  EFI_STATUS            Status    = 0;
+
+  IoWrite8 (HWMON_INDEX_ADDRESS, Register);
+  *Value = IoRead8 (HWMON_DATA_ADDRESS);
+  return Status;
+}
```
Comment (sindhuragrandhi) [inline Platform/AlderlakeBoardPkg/Library/Stage2BoardInitLib/SioChip.c:763]: Newline may be required.

File: Platform/AlderlakeBoardPkg/Library/Stage2BoardInitLib/SioChip.c [hunk 3]
```diff
@@ -0,0 +1,763 @@
...
+It8659HwMonReadRegister (
+  IN  UINT8     Register,
+  OUT UINT8    *Value
+  )
+{
+  EFI_STATUS            Status    = 0;
+
+  IoWrite8 (HWMON_INDEX_ADDRESS, Register);
+  *Value = IoRead8 (HWMON_DATA_ADDRESS);
+  return Status;
```
Comment (gdong1) [inline Platform/AlderlakeBoardPkg/Library/Stage2BoardInitLib/SioChip.c:762]: Maybe just return EFI_SUCCESS.

File: Platform/AlderlakeBoardPkg/Library/Stage2BoardInitLib/SioChip.c [hunk 4]
```diff
@@ -0,0 +1,763 @@
...
+  // been done before this driver is started to manage the device.
+  //
+
+  //
+  // Use S3BootScriptSaveIoWrite8(...) function. This will simply add the
+  // required IO to the S3 table.
+  //
+  DEBUG ((DEBUG_INFO, "SioInitAndSaveS3 Entry\n"));
+
+  //mIoWrite8        = S3BootScriptSaveIoWrite8;
```
[the remark this answers is not in the document]
Comment (koktong-ong) [reply Platform/AlderlakeBoardPkg/Library/Stage2BoardInitLib/SioChip.c:647]: currently SBL is not supporting S3 boot script and skip for this time
will find another way to save the sio config for S3

**Comment by gdong1** [comment]
Need fix the build failure.

**Comment by koktong-ong** [comment]
Build issue fixed

**Review by gdong1 — APPROVED** [review]

**Pull request merged**